[git-webkit] Ignore invalid contributor in mapping

JonWBedard opened the pull request:
#### e1a29e7ecaf64e939b4abcfaf25960b18e1d65a8
<pre>
[git-webkit] Ignore invalid contributor in mapping
<a href="https://bugs.webkit.org/show_bug.cgi?id=250130">https://bugs.webkit.org/show_bug.cgi?id=250130</a>
rdar://103913539

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/contributor.py:
(Contributor.Mapping.add): Don&apos;t add contributor if one can&apos;t be created.
(Contributor.Mapping.__iter__): Skip empty contributors.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e1a29e7ecaf64e939b4abcfaf25960b18e1d65a8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/6/builds/102712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11832 "Built successfully") | [❌ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35738 "Failed to checkout and rebase branch from PR 8235") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/111968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/172209 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106678 "Passed tests") | [❌ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12844 "Failed to checkout and rebase branch from PR 8235") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2740 "Built successfully") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109658 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108488 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/76/builds/12844 "Failed to checkout and rebase branch from PR 8235") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/43/builds/35738 "Failed to checkout and rebase branch from PR 8235") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/94968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/106231 "Passed tests") | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/76/builds/12844 "Failed to checkout and rebase branch from PR 8235") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/43/builds/35738 "Failed to checkout and rebase branch from PR 8235") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/94968 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/5290 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/43/builds/35738 "Failed to checkout and rebase branch from PR 8235") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/82/builds/5439 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2442 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [  ~~🧪 services~~](https://ews-build.webkit.org/#/builders/20/builds/101605 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/11459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/43/builds/35738 "Failed to checkout and rebase branch from PR 8235") | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/79/builds/7181 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3169 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/contributor.py:
```diff
@@ -111,6 +111,9 @@ def add(self, contributor):
             result = self.create(contributor.name, *contributor.emails)
+            if not result:
+                return None
+
```
Why is it OK to silently discard a malformed contributor?

JonWBedard replied:
Other options here are:
- Log an error but continue
- Raise an exception

In practice, I think both of them would be ok, although would prefer "log an error" since I fear we could otherwise find ourselves in a death spiral if someone managed to commit a bad contributor to contributors.json

JonWBedard committed:
[git-webkit] Ignore invalid contributor in mapping
https://bugs.webkit.org/show_bug.cgi?id=250130
rdar://103913539

Reviewed by Elliott Williams.

* Tools/Scripts/libraries/webkitscmpy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/contributor.py:
(Contributor.Mapping.add): Don't add contributor if one can't be created.
(Contributor.Mapping.__iter__): Skip empty contributors.

Canonical link: https://commits.webkit.org/258691@main
